src/console/content.ts

```diff
@@ -533,6 +538,24 @@ class ConsoleContent extends Widget {
     this.prompt.mimetype = this._mimetype;
   }
 
+  protected addContentCell(cell: CodeCellWidget) {
+    this._content.addWidget(cell);
+    const prompt = this.prompt;
+    const index = prompt ? this._cells.indexOf(prompt.model) : -1;
```

Comment:
My confusion stems from this: perhaps a cell model should be added to `_cells` only when it is ready to go into `_content` and not when the prompt is created. That way an `indexOf` is unnecessary, it's always just added to the end of the list. I would propose adding to `_cells` here: https://github.com/R-Brain/jupyterlab/blob/bfb5b04f7b7cfd73d90f7fb6525cfca6b570473c/src/console/content.ts#L441